src/app/cli/src/app.rs

```diff
@@ -135,20 +135,28 @@ pub async fn run(
         cli_catalog.get_one::<GcService>()?.evict_cache()?;
     }
 
-    let result = match cli_commands::get_command(&base_catalog, &cli_catalog, &matches) {
-        Ok(mut command) => {
-            if command.needs_workspace() && !workspace_svc.is_in_workspace() {
-                Err(CLIError::usage_error_from(NotInWorkspace))
-            } else if command.needs_workspace() && workspace_svc.is_upgrade_needed()? {
-                Err(CLIError::usage_error_from(WorkspaceUpgradeRequired))
-            } else if current_account.is_explicit() && !is_multi_tenant_workspace {
-                Err(CLIError::usage_error_from(NotInMultiTenantWorkspace))
-            } else {
-                command.run().await
+    // Wrap command execution in a transaction
+    let result = DatabaseTransactionRunner::run_transactional(
+        &cli_catalog,
+        |cli_catalog_with_transaction| async move {
```

Comment:
Why would this be necessary?
If a particular command, i.e. `kamu login`, requires it, the particular command should control the scope of the transaction.
Right now you are again, wrapping continous commands like `kamu ui` or `kamu system api-server` into a never ending transaction.

Reply:
> Right now you are again, wrapping continous commands like kamu ui or kamu system api-server into a never ending transaction.

oh, yes, you are right...